sympy/solvers/inequalities.py

```diff
@@ -982,3 +986,417 @@ def reduce_inequalities(inequalities, symbols=[]):
+    a problem where it's not possible to satisfy the contraints, you will get this sort of exception.
+    """
+    pass
+
+
+def _pivot(M, i, j):
+    """
+    Example
+    ========
+
+    >>> from sympy.matrices.dense import Matrix
+    >>> from sympy.solvers.inequalities import _pivot
+    >>> m = Matrix([[3,1,5], \
+                    [2,-3,1], \
+                    [0,3,1]])
+    >>> _pivot(m, 1,0) # pivot around 2, which is in row 1 and col 0
+    Matrix([
+    [-3/2, 11/2, 7/2],
+    [ 1/2, -3/2, 1/2],
+    [   0,    3,   1]])
+
+    """
+    if M[i, j] == 0:
+        raise ZeroDivisionError("Tried to pivot about zero-valued entree.")
+    MM = Matrix.zeros(M.rows, M.cols)
+    for ii in range(M.rows):
+        for jj in range(M.cols):
+            if ii == i and jj == j:
+                # The pivot quantity goes into its reciprocal
+                MM[ii, jj] = 1 / M[i, j]
+            elif ii == i and jj != j:
+                #  Entries in the same row as the pivot are divided by the pivot
+                MM[ii, jj] = M[ii, jj] / M[i, j]
+            elif ii != i and jj == j:
+                # Entries in the same column as the pivot are divided by the pivot and changed in sign.
+                MM[ii, jj] = -M[ii, jj] / M[i, j]
+            else:
+                # The remaining entries are reduced in value by the following:
+                # the product of the corresponding entries in the same row and column as themselves and the pivot,
+                # divided by the pivot.
+                MM[ii, jj] = M[ii, jj] - M[ii, j] * M[i, jj] / M[i, j]
+    return MM
+
+
+def _simplex1(M, R, S):
+    """
+    Perform the first phase of the simplex method: find a feasible solution or determine that no feasible solutions
+    exist.
+
+    LINEAR PROGRAMMING: A Concise Introduction by Thomas S. Ferguson
+      - http://web.tecnico.ulisboa.pt/mcasquilho/acad/or/ftp/FergusonUCLA_LP.pdf
+    """
+    rand = random.Random(x=core_random())
+    while True:
+        B = M[:-1, -1]
+        A = M[:-1, :-1]
+        if all(B[i] >= 0 for i in range(B.rows)):
+            return M, R, S # we have found a feasible solution
+
+        for k in range(B.rows):
+            if B[k] < 0:
+                break
+
+        piv_cols = []
+        for j in range(A.cols):
+            if A[k, j] < 0:
+                piv_cols.append(j)
+        if not piv_cols:
+            return None
+        rand.shuffle(piv_cols)
+        j0 = piv_cols[0]
+
+        ratio = B[k] / A[k, j0]
+        piv_rows = [(ratio, k)]
+        for i in range(A.rows):
+            if A[i, j0] > 0 and B[i] > 0:
+                ratio = B[i] / A[i, j0]
+                piv_rows.append((ratio, i))
+
+        piv_rows = sorted(piv_rows, key=lambda x: (x[0], x[1]))
+        piv_rows = [(ratio, i) for ratio, i in piv_rows if ratio == piv_rows[0][0]]
+        rand.shuffle(piv_rows)
+        _, i0 = piv_rows[0]
+
+        M = _pivot(M, i0, j0)
+        R[j0], S[i0] = S[i0], R[j0]
```

Comment:
I'm not sure why you can't just catch errors than having duplicate functions that returns some error code though.
If you don't like catching errors, you can redesign the whole function to return error code or customized error type, or vice versa, 
but it shouldn't be meaningless duplicate like `simplex1, simplex2, ...`

Reply:
The purpose of this change is for the `find_feasible` function. It's not necessary to perform the entire simplex method if all you're interested in is checking if a system of inequalities is feasible. 

Checking if a system of inequalities is feasible is relevant to my GSoC work with the new assumptions.

Reply:
Exceptions or early returns are able to short-circuit such cases anyway.

Reply:
By early return you mean it would be a better idea to add a parameter to `_simplex` that would cause an early return on the first feasible solution instead of splitting it up into two methods?

Reply:
It would be better to refactor with respect to the actual duplicate code like the while loop, but the point is not to complicate the control sequence (too much optional parameters) or not to introduce any code that doesn't look obvious.
I don't answers for them now, but it can be part of your own responsibility.